roles/cloud-ec2/tasks/prompts.yml

```diff
@@ -53,3 +53,28 @@
         [{{ default_region }}]
     register: _algo_region
   when: region is undefined
+
+- block:
+  - name: Get existing available Elastic IPs
+    ec2_elasticip_facts:
+    register: raw_eip_addresses
+
+  - set_fact:
+      available_eip_addresses: "{{ raw_eip_addresses.addresses | selectattr('association_id', 'undefined') | list }}"
+
+  - pause:
+      prompt: |
+        Do you want to use a pre-allocated Elastic IP?
+        (leave blank to allocate a new IP)
+          {% for eip in available_eip_addresses %}
+          {{ loop.index }}. {{ eip['public_ip'] }}
+          {% endfor %}
+          ['']
+    register: _use_existing_eip
+
+  - set_fact:
+      use_existing_eip: >-
+        {% if _use_existing_eip.user_input is defined and _algo_region.user_input != "" %}
+        {{ available_eip_addresses[_use_existing_eip.user_input | int -1 ]['allocation_id'] }}
+        {%- else %}{% endif %}
+  when: use_existing_eip is undefined
```

Comment:
Can we run this block only when `use_existing_eip` is set to true in config.cfg? By default it should be false

Reply:
Sure, that makes a lot of sense! I just pushed a commit changing this.

Reply:
Thanks. I guess the installation will fail, because there is no default for this variable, so you also need to add `use_existing_eip: false` to config.cfg, and add a description. Also, I see that you use the same variable in `set_fact`, that probably needs to be renamed to get it more understandable

Reply:
Thanks! I want to cooperate, but I guess I don't understand how you want the behavior to change. Are you also wanting the prompt asking if you want to use an existing EIP to be hidden if use_existing_eip is false in config.cfg?

Reply:
No, I want `use_existing_eip` to be set to `false` in config.cfg, and the prompt asking only if it's set to `true`